common/zeroconf/zeroconf-dashelhub.h

```diff
@@ -0,0 +1,41 @@
+/*
+	Aseba - an event-based framework for distributed robot control
+	Copyright (C) 2007--2016:
+		Stephane Magnenat <stephane at magnenat dot net>
+		(http://stephane.magnenat.net)
+		and other contributors, see authors.txt for details
+
+	This program is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Lesser General Public License as published
+	by the Free Software Foundation, version 3 of the License.
+
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Lesser General Public License for more details.
+
+	You should have received a copy of the GNU Lesser General Public License
+	along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <dashel/dashel.h>
+#include "zeroconf.h"
+
+namespace Aseba
+{
+	//! Use the Dashel run loop to watch the DNS Service for updates, and trigger callback
+	//! processing as necessary.
+	class DashelhubZeroconf : Zeroconf, Dashel::Hub
+	{
+	protected:
+		//! Set up function called after a discovery request has been made. The file
+		//! descriptor associated with zdr.serviceref must be watched, to know when to
+		//! call DNSServiceProcessResult, which in turn calls the callback that was
+		//! registered with the discovery request.
+		virtual void processDiscoveryRequest(ZeroconfDiscoveryRequest & zdr);
+
+	private:
+		//! Collection of (tcp:) SocketStreams that watch the serviceref file descriptors.
+		std::vector<Dashel::Stream *> zeroconfStreams;
```

Comment:
The streams provided by the base `Dashel::Hub` may be any streams. We need to remember which ones we created to listen to `ZeroconfDiscoveryRequest` sockets, so that in `incomingData` we can call `DNSServiceProcessResult` on the right servicerefs.

So should really be `std::map<Dashel::Stream *, ZeroconfDiscoveryRequest &>`